ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LedgerPeers/Type.hs

```diff
@@ -27,6 +28,15 @@ import GHC.Generics
 import NoThunks.Class
 import Ouroboros.Network.PeerSelection.RelayAccessPoint (RelayAccessPoint)
 
+-- | Peer snapshot type used to serialize peers from the
+-- current state of a node's ledger:
+-- LocalStateQuery app server running in ouroboros-consensus-diffusion
+-- receives a query (eg. cardano-cli) demanding a dump of this data
+-- from the current ledger. Consensus uses CBOR.
+newtype LedgerPeerSnapshot = LedgerPeerSnapshot
+  { unLedgerPeerSnapshot :: (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))]) }
+  deriving newtype (Show, ToCBOR, FromCBOR)
```

Comment:
hmm I see, I wonder if getting those from cardano-binary is okay, at the time I was working with peer sharing I remember we were striving to avoid automatically derived encoders for the reasons I said above. Not sure if the situation changed. cc @coot

Reply:
The WithOrigin and SlotNo types have automatically derived instances of Serialise, and then in cardano-slotting there's a hand-cranked ToCBOR and FromCBOR that calls out to encode/decode. I suppose I could propose a patch there so that we have manual instances for everything.